wails-frontend/src/pages/passwordPrompt.tsx

```diff
@@ -52,78 +53,84 @@ function PasswordPrompt() {
     }
   }
 
-  function isDisabled() {
-    if (isNewPassword || isImport) {
-      return !(password && passwordConfirm);
+  const [error, setError] = useState<ErrorObject | null>(null);
+
+  function validate(e: SyntheticEvent) {
+    const form = parseForm(e);
+    const { password, passwordConfirm } = form;
+
+    if (!hasMoreThanFiveChars(password.toString())) {
+      setError({ password: 'Password must have at least 5 characters' });
+      return false;
+    } else if (
+      !hasSamePassword(password.toString(), passwordConfirm.toString())
+    ) {
+      setError({ password: "Passwords don't match" });
+      return false;
     }
-    return !password;
+    return true;
   }
 
-  function applyPassword() {
-    if (password.length) {
-      if (passwordConfirm && password !== passwordConfirm) {
-        setResultMsg("Passwords don't match");
-        return;
-      }
-      EventsOnce(
-        events.passwordResult,
-        handleApplyResult(
-          navigate,
-          req,
-          setResultMsg,
-          state.req.Action === promptAction.importReq,
-        ),
-      );
-
-      if (state.req.Action === promptAction.importReq) {
-        return ImportPrivateKey(state.pkey, state.nickname, password);
-      }
-      return ApplyPassword(password);
+  function save(e: SyntheticEvent) {
+    const form = parseForm(e);
+    const { password } = form;
+
+    EventsOnce(
+      events.passwordResult,
+      handleApplyResult(navigate, req, setError, isImport),
+    );
+
+    return isImport
+      ? ImportPrivateKey(state.pkey, state.nickname, password.toString())
+      : ApplyPassword(password.toString());
+  }
+
+  async function handleSubmit(e: SyntheticEvent) {
+    e.preventDefault();
+
+    if (validate(e)) {
+      save(e);
     }
   }
 
   return (
     <div className="bg-primary min-h-screen">
-      <div
-        className="flex flex-col justify-center h-screen
-        max-w-xs min-w-fit text-f-primary m-auto"
-      >
-        <h1 className="mas-title">{req.Msg}</h1>
-        <p className="mas-body pt-4">{getSubtitle()}</p>
-        <div className="pt-4">
-          <Password
-            onChange={(e) => setPassword(e.target.value)}
-            value={password}
-            name="input"
-            placeholder="Enter your password"
-          />
-        </div>
-        {(isNewPassword || isImport) && (
+      <form ref={form} onSubmit={handleSubmit}>
+        <div
+          className="flex flex-col justify-center h-screen
+          max-w-xs min-w-fit text-f-primary m-auto"
+        >
+          <h1 className="mas-title">{req.Msg}</h1>
+          <p className="mas-body pt-4">{getSubtitle()}</p>
           <div className="pt-4">
-            <Password
-              onChange={(e) => setPasswordConf(e.target.value)}
-              value={passwordConfirm}
-              name="input"
-              placeholder="Confirm your password"
+            <Input
```

Comment:
I guess you can use `<Password>` here. The API for the error is already merged!

Reply:
Yes sorry! I will correct this now.